File: src/main/java/com/github/zxh/classpy/classfile/attribute/AttributeInfo.java
package com.github.zxh.classpy.classfile.attribute;

import com.github.zxh.classpy.classfile.ClassComponent;

/*
attribute_info {
    u2 attribute_name_index;
    u4 attribute_length;
    u1 info[attribute_length];
}
 */
public abstract class AttributeInfo extends ClassComponent {

    {
        u2("attribute_name_index");
        u4("attribute_length");
    }

}

File: src/main/java/com/github/zxh/classpy/classfile/attribute/AttributeInfo.kt
package com.github.zxh.classpy.classfile.attribute;

import com.github.zxh.classpy.classfile.ClassComponent;

/*
attribute_info {
    u2 attribute_name_index;
    u4 attribute_length;
    u1 info[attribute_length];
}
 */
open class AttributeInfo : ClassComponent() {

    init {
        u2("attribute_name_index");
        u4("attribute_length");
    }

}
